File: app/src/main/java/in/gov/mahapocra/farmerapppks/ui/adapters/ClimateGridAdapter.java
package in.gov.mahapocra.farmerapppks.ui.adapters;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.squareup.picasso.MemoryPolicy;
import com.squareup.picasso.NetworkPolicy;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;

import in.gov.mahapocra.farmerapppks.R;
import in.gov.mahapocra.farmerapppks.activity.DiseaseInformation;
import in.gov.mahapocra.farmerapppks.data.model.ClimateGridModel;

public class ClimateGridAdapter extends ArrayAdapter<ClimateGridModel> {

    String image_url;
    private Context mContext;
    String activity;



        public ClimateGridAdapter(@NonNull Context context, ArrayList<ClimateGridModel> ClimateGridModelArrayList, String activity) {
            super(context, 0, ClimateGridModelArrayList);
            this.mContext = context;
            this.activity = activity;
        }

        @NonNull
        @Override
        public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
            View listitemView = convertView;
            if (listitemView == null) {
                // Layout Inflater inflates each item to be displayed in GridView.
                listitemView = LayoutInflater.from(getContext()).inflate(R.layout.single_climate_grid, parent, false);
            }
            ClimateGridModel ClimateGridModel = getItem(position);
            TextView climateName = listitemView.findViewById(R.id.climate_name);
            RelativeLayout gridItem = listitemView.findViewById(R.id.gridItem);
            ImageView climateImage = listitemView.findViewById(R.id.climate_image);
            climateName.setText(ClimateGridModel.getClimate_name());
            image_url= ClimateGridModel.getClimate_image();
            loadImage(climateImage,image_url);

            if(activity.equalsIgnoreCase("PestAndDiseasesAdpater")){
                gridItem.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        Intent intent = new Intent(ClimateGridAdapter.this.getContext(), DiseaseInformation.class);
                        //String Id = ClimateGridModel.getWebUrl();
                        intent.putExtra("id",Integer.parseInt(ClimateGridModel.getWebUrl()));
                        String currentString = ClimateGridModel.getClimate_name();
                        String[] separated = currentString.split("\n");
                        intent.putExtra("name",separated[1]);
                        mContext.startActivity(intent);
                    }
                });
            }
            return listitemView;

        }

    public static void loadImage(ImageView climateImage, String image_url) {
        try {
            Picasso.get().invalidate(image_url);
            Picasso.get()
                    .load(image_url)
                    .memoryPolicy(MemoryPolicy.NO_CACHE, MemoryPolicy.NO_STORE)
                    .networkPolicy(NetworkPolicy.NO_CACHE, NetworkPolicy.NO_STORE)
                    .placeholder(R.drawable.ic_thumbnail)
                    .resize(500, 400)
                    .centerCrop()
                    .error(R.drawable.ic_thumbnail)
                    .into(climateImage);
        } catch (Exception ex) {
            ex.toString();
        }

    }
}

File: app/src/main/java/in/gov/mahapocra/mahavistaarai/ui/adapters/ClimateGridAdapter.kt
package `in`.gov.mahapocra.mahavistaarai.ui.adapters

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.cardview.widget.CardView
import com.squareup.picasso.MemoryPolicy
import com.squareup.picasso.NetworkPolicy
import com.squareup.picasso.Picasso
import `in`.gov.mahapocra.mahavistaarai.R
import `in`.gov.mahapocra.mahavistaarai.data.model.ClimateGridModel
import `in`.gov.mahapocra.mahavistaarai.ui.screens.dashboard.menugrid.pest.DiseaseInformation

class ClimateGridAdapter(
    private val mContext: Context,
    climateGridModelArrayList: ArrayList<ClimateGridModel>,
    var activity: String
) : ArrayAdapter<ClimateGridModel?>(mContext, 0, climateGridModelArrayList) {
    var imageUrl: String? = null

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var listItemView = convertView
        if (listItemView == null) {
            // Layout Inflater inflates each item to be displayed in GridView.
            listItemView = LayoutInflater.from(context)
                .inflate(R.layout.single_climate_grid, parent, false)
        }
        val climateGridModel: ClimateGridModel? = getItem(position)
        val climateName = listItemView.findViewById<TextView>(R.id.climate_name)
        val gridItem = listItemView.findViewById<CardView>(R.id.gridItem)
        val climateImage = listItemView.findViewById<ImageView?>(R.id.climate_image)
        climateName.text = climateGridModel?.climate_name
        imageUrl = climateGridModel?.climate_image
        loadImage(climateImage, imageUrl)

        if (activity.equals("PestAndDiseasesAdpater", ignoreCase = true)) {
            gridItem.setOnClickListener { v: View ->
                val intent = Intent(v.context, DiseaseInformation::class.java)
                intent.putExtra("id", climateGridModel?.webUrl?.toInt())
                val currentString: String = climateGridModel?.climate_name ?: ""
                val separated: Array<String?> =
                    currentString.split("\n".toRegex()).dropLastWhile { it.isEmpty() }
                        .toTypedArray()
                intent.putExtra("name", separated[1])
                mContext.startActivity(intent)
            }
        }
        return listItemView
    }

    companion object {
        fun loadImage(climateImage: ImageView?, image_url: String?) {
            try {
                Picasso.get().invalidate(image_url)
                Picasso.get()
                    .load(image_url)
                    .memoryPolicy(MemoryPolicy.NO_CACHE, MemoryPolicy.NO_STORE)
                    .networkPolicy(NetworkPolicy.NO_CACHE, NetworkPolicy.NO_STORE)
                    .resize(500, 400)
                    .centerCrop()
                    .into(climateImage)
            } catch (ex: Exception) {
                ex.toString()
            }
        }
    }
}
